lib/npg_tracking/data/transcriptome/find.pm

```diff
@@ -2,13 +2,34 @@ package npg_tracking::data::transcriptome::find;
 
 use Moose::Role;
 use Carp;
-
 use npg_tracking::util::abs_path qw(abs_path);
 
+Readonly::Scalar our $DEFAULT_ALIGNER => q[tophat2];
+Readonly::Scalar our $DEFAULT_QUANTIFIER => q[salmon];
+Readonly::Scalar our $DEFAULT_ANALYSIS => $DEFAULT_ALIGNER;
+
 with qw/ npg_tracking::data::reference::find /;
 
 our $VERSION = '0';
 
+has 'analysis' => (default  => $DEFAULT_ANALYSIS,
+                   is       => 'ro',
+                   required => 0,);
+
+has '_index_name_ext' => (isa      => q{Maybe[Str]},
+                          is       => q{ro},
+                          required => 0,
+                          lazy     => 1,
+                          builder  => q{_build_index_name_ext},);
+
+sub _build_index_name_ext {
+    my $self = shift;
+    ## which type of file to look for in the index folder based on the analysis
+    my $ext = $self->analysis eq $DEFAULT_ALIGNER ? q[bt2] :
```

Comment:
There maybe an opportunity to improve the semantics here and also w.r.t. using `DEFAULT_`